.github/workflows/deploy.yml

```
@@ -36,11 +36,22 @@ jobs:
       Version: latest
     secrets: inherit
 
+  approval-step:
+    # This step is here just to get approval, without it GHA will proceed onto
+    # the next steps, check the concurrency rules and cancel them in certain
+    # cases. The resolution order appears to be:
+    # 1. Move to next job
+    # 2. Check concurrency (cancelling here is not what we want)
+    # 3. Ask for approval
+    needs: qa-us-west-1
+    with:
+      Environment: prod
```

Should this be:

```suggestion
      Environment: approval
```

And move the requirement for approval to this environment, not prod?

---

And your thinking is we remove the protection rules against the `prod` environment and add them to `approval`?

---

If this is what you're thinking of doing then it should work, but would be "unconventional" usage of environments. As long as we're documenting why we're doing this I think we can go with it. It might be a good idea to ping GitHub support and tell them we're having to use this workaround in case they have other ideas.

---

Hmm. The fake `approval` environment would show up in various places in the GitHub UI. In the app's list of environments in the sidebar, on the app's deployment history page, etc. It's gonna say things like "**Deployed to approval**" in the UI.

A fake `approval` environment may potentially be the lesser of two evils compared to using the `prod` environment for the approval job though because I don't think there should _be_ a single `prod` environment in the first place (see my notes about this below)